packages/fx/README.md

```diff
@@ -1,78 +1,97 @@
+
+## Trigger execution from an action
+
+The `fx.execute` function is the starting point of a chain of operators that the Fx library can handle. It works as follows:
+
+* it takes as argument a function that is to be executed when the effect is triggered (this function must return an observable)
+* if the function throws an error, it will be caught and the function will return a `Failure` object in the stream, wrapping both the input and the error
+* if the function succeeds, it will return a `Success` object in the stream, wrapping both the input and the output
+* therefore, the `fx.execute` never throws an error and always returns either a `Success` or a `Failure`
 
 ```ts
 constructore(private fx: FxService) {}
 
 myEffect$ = createEffect(
   () => ofType(myAction).pipe(
     this.fx.execute(action => ...),
-    tap(({ input, output }) => ...) // the result is wrapping the input and the output
+    tap((result: Success | Failure) => ...)
     ...
   )
 );
 ```
 
-### Ask user for confirmation
+After the `fx.execute` you can use your own operators or more `fx` built-in methods.
 
-Before executing anything, easily ask your user to confirm! If the user rejects, the whole stream is stopped.
+## Notify on success/failure
 
 ```ts
-constructore(private fx: FxService) {}
-
 myEffect$ = createEffect(
   () => ofType(myAction).pipe(
-    this.fx.confirmFilter(action => ({ ... })), // pass the data to your confirmation UI handler
-    this.fx.execute(action => ...), // only executes if the user confirms
+    this.fx.execute(action => ...), // EXECUTION
+    this.fx.notifySuccess(({ input, output }) => ({ ... })), // will only run if the execution succeeds, generate data to pass to the notification UI handler
+    this.fx.notifyFailure(({ input, error }) => ({ ... })), // will only run if the execution fails, generate data to pass to the notification UI handler
     ...
   )
 );
 ```
 
-You can customize the confirmation UI by providing a confirmation service when initializing the module.
+You can customize the notification UI by providing a notifier service when initializing the module.
 
 ```ts
-import { FxModule, Confirm } from '@iwerk/fx';
+import { FxModule, Notifier } from '@iwerk/fx';
 
 @Injectable()
-class CustomConfirm implements Confirm {
-
+class CustomNotifier implements Notifier {
+  openSuccess<InputType, OutputType>(
+    data: unknown,
+    success: Success<InputType, OutputType>
+  ) {
+    // display a success notification
+  }
+
+  openFailure<InputType>(data: unknown, success: Failure<InputType>) {
+    // display a failure notification
+  }
 }
 
 @NgModule({
   imports: [
     ...
     FxModule.forRoot({
-      confirm: {
-        useClass: CustomConfirm
+      notifier: {
+        useClass: CustomNotifier
```

Comment:
Why is it not a separate provider that can be easily replaced (in tests)? Can you use `useValue`?

Reply:
The type is `Omit<Provider, 'provide'>`. You can therefore use the standard provider declaration, useValue, useExisting, useClass, useFactory etc.